wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -588,6 +609,143 @@ private String inferEngine() throws ModelNotFoundException {
+        URI configUri = null;
+        if (modelId != null && modelId.startsWith("s3://")) {
+            // This is definitely suboptimal, but for the majority of cases we need to download this
+            // s3 model eventually, so it is not the worst thing to download it now.
+            downloadS3();
+            if (Files.isRegularFile(downloadS3Dir.resolve("config.json"))) {
+                configUri = downloadS3Dir.resolve("config.json").toUri();
+            } else if (Files.isRegularFile(downloadS3Dir.resolve("model_index.json"))) {
+                configUri = downloadS3Dir.resolve("model_index.json").toUri();
+            }
+        } else if (modelId != null) {
+            prop.put("option.modelId", modelId);
+            configUri = URI.create("https://huggingface.co/" + modelId + "/raw/main/config.json");
+            HttpURLConnection configUrl = (HttpURLConnection) configUri.toURL().openConnection();
+            // stable diffusion models have a different file name with the config... sometimes
+            if (HttpURLConnection.HTTP_OK != configUrl.getResponseCode()) {
+                configUri =
+                        URI.create(
+                                "https://huggingface.co/" + modelId + "/raw/main/model_index.json");
+            }
+        } else if (Files.isRegularFile(modelDir.resolve("config.json"))) {
+            configUri = modelDir.resolve("config.json").toUri();
+        } else if (Files.isRegularFile(modelDir.resolve("model_index.json"))) {
+            configUri = modelDir.resolve("model_index.json").toUri();
+        }
+        return configUri;
+    }
+
+    private JsonObject getHuggingFaceModelConfig() throws ModelException, IOException {
+        String modelId = Utils.getEnvOrSystemProperty("HF_MODEL_ID");
+        if (modelId == null) {
+            modelId = prop.getProperty("option.model_id");
+        }
+        if (modelId == null) {
+            // Deprecated but for backwards compatibility
+            modelId = prop.getProperty("option.s3url");
+        }
+        URI modelConfigUri = generateHuggingFaceConfigUri(modelId);
+        if (modelConfigUri == null) {
+            return null;
+        }
+        try (InputStream is = modelConfigUri.toURL().openStream();
+                BufferedReader reader =
+                        new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
+            return JsonUtils.GSON.fromJson(reader, JsonElement.class).getAsJsonObject();
+        } catch (IOException e) {
+            logger.error(
+                    "The huggingface hub model_id {} is not valid. Please verify this is a valid"
+                            + " model_id",
+                    modelId);
+            throw e;
+        }
+    }
+
+    private boolean isFasterTransformerRecommended(String modelType) {
+        return "t5".equals(modelType);
+    }
+
+    private boolean isDeepSpeedRecommended(String modelType) {
```

Comment:
we probably want to build mechanism to check if deepspeed is installed or not.

Reply:
We would need something for FT as well right?

I was thinking of this in the context of the HuggingFaceModel from SM PySDK. With separate images for DS and FT, we would be locked into an environment already by the time we're here...

You're right this needs to be reworked...

Reply:
This mechanism should execute in a standalone mode, not relate to any server. Which is we should be able to detect if certain package is installed or not

Reply:
done